### DISCUSSION TITLE
Current layout info

### DISCUSSION THREAD

**Discussion by di-effe** [leftwm/leftwm#603] [Q&A] [answered]
Hi, first of all: love the project. I tried most of the WM around and never felt 100% sure to jump in the tiling world.
And now I am tinkering with a custom theme for hours :D

That said and possibly due to fatigue, I can´t find a way to display (possibly in polybar) the layout in use.
Is it possible?

I saw an Epitaph screenshot showing it but I can´t find that specific conf even in VentGrey repository
https://github.com/VentGrey/Epitaph/blob/master/screenshots/01.png

Thanks

**Comment by VuiMuich** [comment, marked as answer]
Welcome Adventurer!

There are a bunch of ways:
- you can use `leftwm state -w0 -s "{{workspaces.layout}}"`
- or just `leftwm state` and parse the JSON it outputs 'manually'
- our if you are using a `template.liquid`you could try to add something like
```liquid
{% for layout in workspaces %}
%{F#000000}%{B#E60053}  {{workspace.index.layout}}  %{B-}%{F-}
{% endfor %}
```
Not sure though if this liquid syntax is correct and can't test right now.

**Comment by di-effe** [comment]
Thanks, after my post I got back to thub.com/leftwm/leftwm/wiki/Customizing-Themes-with-%60liquid%60-templates and find it there, looking at my face. I guess I spent too much time in the terminal today :D

I got it showing up in both my workspaces using two modules like 

```
[module/layout0]
type = custom/script
exec = leftwm-state -w 0 -s "{{ workspace.layout }}"
tail = true
```

and

```
[module/layout1]
type = custom/script
exec = leftwm-state -w 1 -s "{{ workspace.layout }}"
tail = true
```

would be possible to catch the workspace ID automatically?
something like

`exec = leftwm-state -w {WORKSPACE_ID} -s "{{ workspace.layout }}"`

**Reply by VuiMuich** [reply]
not shure, if polybar is able to expand this before executing.

**Reply by di-effe** [reply]
Yeah, looks like it can´t, but I will dig a little

**Comment by di-effe** [comment]
Not too shabby for my first attemp
![Screenshot_10](https://user-images.githubusercontent.com/72241321/144765577-ef2e4d6b-a803-47b4-8dd7-76f07ca95531.png)

**Reply by mautamu** [reply]
Ooh that looks neat!

**Reply by di-effe** [reply]
So nice of you :)

Btw with a night sleep it was soo much easier using the template.liquid.
But ehy.. now I got both solutions to chose from :D
